File: AN_2/SEM_1/MAP/lab2/src/decorator/StrategyTaskRunner.java
package decorator;

import container.Container;
import model.Task;
import factory.TaskContainerFactory;
import Enum.Strategy;

public class StrategyTaskRunner implements TaskRunner {

    private Container container;

    public StrategyTaskRunner(Strategy strategy)
    {
        this.container = TaskContainerFactory.getInstance().createContainer(strategy);
    }

    @Override
    public void executeOneTask()
    {
        Task task = container.remove();
        if (task != null)
            task.execute();
    }

    @Override
    public void executeAll()
    {
        while (hasTask()) {
            executeOneTask();
        }
    }

    @Override
    public void addTask(Task t)
    {
        container.add(t);
    }

    @Override
    public boolean hasTask()
    {
        return !container.isEmpty();
    }
}

File: AN_2/SEM_1/MAP/lab9/lab9/Decorator/StrategyTaskRunner.cs
using Container;
using Model;
using Factory;
using Enum;

namespace Decorator
{
    public class StrategyTaskRunner : ITaskRunner
    {
        private IContainer container;

        public StrategyTaskRunner(Strategy strategy)
        {
            this.container = TaskContainerFactory.GetInstance().CreateContainer(strategy);
        }

        public void ExecuteOneTask()
        {
            Tasks task = container.Remove();
            if (task != null)
                task.Execute();
        }

        public void ExecuteAll()
        {
            while (HasTask())
            {
                ExecuteOneTask();
            }
        }

        public void AddTask(Tasks t)
        {
            container.Add(t);
        }

        public bool HasTask()
        {
            return !container.IsEmpty();
        }
    }
}
